content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,136 @@
+Peers subscribed to a spam-protected `pubsubTopic` are only allowed to send one message per `epoch`
+(messaging rate MAY be enforced for `WakuMessages` with a specific `contentTopic` published on a `pubsubTopic`, but that is an implementation choice).
+
 ## SetUp and Registration
-A peer willing to publish a message is required to register. 
+Peers subscribed to a specific `pubsubTopic` form a [rln group](/spec/32).
+<!-- link to the rln group definition in the rln RFC -->
+Peers MUST be registered to the rln group to be able to publish messages.
 Registration is moderated through a smart contract deployed on the Ethereum blockchain. 
-The state of the contract contains the list of registered members. 
+Each peer has an [rln key pair](/spec/32) denoted by `sk`  and `pk`.
+The `sk` is secret data and MUST be persisted securely by the peer.
+The state of the membership contract contains the list of registered members' public identity keys i.e., `pk`s. 
+For the registration, a peer creates a transaction that invokes the registration function of the contract via which registers its `pk` in the group. 
+The transaction also transfers  `v` amount of ether to the contract to be staked. 
+`v` is a system parameter.
+The peer who has the "private key" `sk` associated with a registered `pk` would be able to withdraw a portion `p` of the staked fund by providing valid proof. <!-- a secure way to prove the possession of a pk is yet under discussion, maybe via commit and reveal -->
+`p` is also a system parameter.
+
+Note that  `sk` is initially only known to its owning peer however, it may get exposed to other peers in case the owner attempts spamming the system i.e., sending more than one message per `epoch`.
 An overview of registration is illustrated in Figure 1.
 
-For the registration, a peer creates a transaction that sends x (TODO to be specified) ETH to the contract. 
-The peer who has the "private key" `sk` associated with that deposit would be able to withdraw x ETH by providing valid proof. 
-Note that  `sk` is initially only known by the owning peer however it may get exposed to other peers in case the owner attempts spamming the system i.e., sending more than one message per epoch.
+![Figure 1: Registration.](../../../../rfcs/17/rln-relay.png)
 
-<!-- TODO: the interaction with the contract is subject to change depending on the final implementation -->
 
-<!-- Once registered, the peer obtains the root of the tree (after the registration of the current peer) i.e., `root`  as well as the authenticity path `authPath`. A peer can prove her membership using the `authPath`. -->
+## Publishing
 
+To publish at a given `epoch`, the publishing peer proceeds based on the regular [`11/WAKU2-RELAY`](/spec/11) protocol.  
+However, to protect against spamming, each `WakuMessage` (which is wrapped inside the `data` field of a PubSub message) MUST carry a [`RateLimitProof`](##RateLimitProof) with the following fields. 
```

Comment:
Paragraph should have line in between (same across document)

Reply:
Yes, they have, I cannot see where the line bw paragraphs are missing?